README.md

```diff
@@ -1,65 +1,82 @@
+
+The processing functions defined in the user's nodes accept a packet to be processed as a parameter. After they are done processing a packet, packets can be sent onto the next channel (one that the user node has knowledge of) in the pipeline by acquiring a packet to be filled from that channel and filling it with the processed data. The filled packet is automatically queued for processing in the next channel upon the destruction of the stub beloning to the filled packet. The space in memory belonging to the processed packet is freed up for reuse automatically upon the destruction of the stub belonging to the packet.
 
+At any point after the scheduler was started, it may be stopped, which means it will instruct all worker threads to finish their processing, wait for them to complete their current tasks, and re-enter the setup/preparation phase.
+
+For a more detailed description on how each component in the system works, please check out our documentation, which you can build using Doxygen following the steps outlined below.
 
 ## Normal Build for Your Project
 
 You need the following packages installed on your system to build the library:
- - `git` for downloading this repo
- - `cmake` and `ninja-build` for running the build
- - `gcc`, `g++` for compiling with GCC _OR_
- - `clang`, `clang++` and `llvm` for compiling with Clang
-
-Optional:
- - `doxygen` will generate HTML documentation for you
+  - `git` for downloading this repo
+  - `cmake` and `ninja-build` to make and use the build system
+  - `gcc`, `g++` for compiling with GCC _OR_
+  - `clang`, `clang++` and `llvm` for compiling with Clang
+  - \[optional\] `doxygen` to generate HTML documentation
 
 Steps to build:
- 1. Download this repo
- 2. Run `./build.sh`
- 3. What you need for your project:
-   - `include` folder
-   - `build/libpipert.so` dynamic library _OR_
-   - `build/libpipert_static.a` static library
- 4. Find the documentation under `build/docs/index.html`
-
+1. Download the repo using `git clone https://github.com/gerazo/pipert.git`
+2. Run `./build.sh` to build the static and dynamic library in release mode
+3. What you need for your project:
+    - an `include` folder
```

Comment:
Here, the developer needs our include folder, not a random one. This was hazy in the original as well.

Reply:
Yeah, I wasn't sure about that, and I only suspected it vaguely. Thanks for clarifying.